parser-typechecker/src/Unison/Codebase/FileCodebase.hs

```diff
@@ -292,13 +293,16 @@ pushGitRootBranch
   :: MonadIO m
   => Codebase.SyncToDir m
   -> Branch.Cache m
+  -> PushOnEmptyDest
   -> Branch m
   -> RemoteRepo
   -> SyncMode
   -> ExceptT GitError m ()
-pushGitRootBranch syncToDirectory cache branch repo syncMode = do
+pushGitRootBranch syncToDirectory cache onEmptyDest branch repo syncMode = do
   -- Pull the remote repo into a staging directory
   (remoteRoot, remotePath) <- viewRemoteBranch' cache (repo, Nothing, Path.empty)
+  when (remoteRoot == Branch.empty && onEmptyDest == AbortOnEmptyDest) $
+    throwError $ GitError.PushDestinationIsEmpty repo
```

Comment:
This is checking if the repo root is empty, but I think we instead want it to check the target sub-namespace before we get in here.